packages/wdio-cli/tests/utils.test.ts

```diff
@@ -391,19 +391,54 @@ describe('getCapabilities', () => {
     })
 
     it('should return driver with capabilities for ios', () => {
-        expect(getCapabilities({ option: 'foo.app', deviceName: 'fooName', udid: 'num', platformVersion: 'fooNum' })).toMatchSnapshot()
+        expect(getCapabilities({ option: 'foo.app', deviceName: 'fooName', udid: 'num', platformVersion: 'fooNum' } as any)).toMatchSnapshot()
         expect(getCapabilities({ option: 'ios' } as any)).toMatchSnapshot()
     })
 
     it('should return driver with capabilities for desktop', () => {
         expect(getCapabilities({ option: 'chrome' } as any)).toMatchSnapshot()
     })
+
+    it('should throw config not found error', () => {
+        jest.unmock('@wdio/config')
+        jest.resetModules()
+        const getCapabilities = require('../src/utils').getCapabilities
+        expect(()=>getCapabilities({ option: './test.js', capabilities: 2 } as any)).toThrowErrorMatchingSnapshot()
+    })
+
+    it('should throw capability not provided', () => {
+        jest.unmock('@wdio/config')
+        jest.resetModules()
+        const getCapabilities = require('../src/utils').getCapabilities
+        expect(()=>getCapabilities({ option: './packages/wdio-cli/tests/__testData__/wdio.conf.js' } as any)).toThrowErrorMatchingSnapshot()
+    })
+
+    it('should through capability not found', () => {
+        jest.unmock('@wdio/config')
+        jest.resetModules()
+        const getCapabilities = require('../src/utils').getCapabilities
+        expect(()=>getCapabilities({ option: 'packages/wdio-cli/tests/__testData__/wdio.conf.js', capabilities: 5 } as any)).toThrowErrorMatchingSnapshot()
+    })
+
+    it('should get capability from wdio.conf.js', () => {
+        jest.unmock('@wdio/config')
```

Comment:
I don't think it is a good idea to actually use the real configparser for these tests. Let's mock them out by importing the mock at the top of the file:
```ts
import { ConfigParser } from '@wdio/config'
```
and then specify functionalities of certain methods, e.g.:
```ts
jest.spyOn(ConfigParser.prototype, 'getCapabilities').mockReturnValue([{ browserName: 'chrome' }]);
```

This will make it easier to test all error cases for your if statement. Let me know if you have any questions to that.

Reply:
The `addConfigFile` method is already mocked here: https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-config/tests/__mocks__/@wdio/config.ts#L10. There should be no need to mock it. What error are you seeing?